cloudify_cli/commands/logs.py

```diff
@@ -79,42 +81,54 @@ def download(output_path, all_nodes, logger):
-                            .format(node['manager_ip']))
+                            .format(node.get('host_ip')))
                 continue
 
             if not output_path:
                 output_path = os.getcwd()
-            host_string = env.build_manager_host_string(
-                ssh_user=ssh_user, ip=node['manager_ip'])
-            archive_path_on_manager = _archive_logs(
-                logger,
-                host_string=host_string,
-                key_filename=ssh_key)
+            host_string = env.build_host_string(ip=node.get('host_ip'),
+                                                ssh_user=ssh_user)
+            archive_path_on_host = _archive_logs(logger,
+                                                 host_string=host_string,
+                                                 key_filename=ssh_key)
             logger.info('Downloading archive to: {0}'.format(output_path))
-            ssh.get_file_from_manager(archive_path_on_manager,
-                                      output_path,
-                                      host_string=host_string,
-                                      key_filename=ssh_key)
-            logger.info('Removing archive from manager...')
-            ssh.run_command_on_manager(
-                'rm {0}'.format(archive_path_on_manager), use_sudo=True,
-                host_string=host_string,
-                key_filename=ssh_key)
+            ssh.get_file_from_host(archive_path_on_host,
+                                   output_path,
+                                   host_string=host_string,
+                                   key_filename=ssh_key)
+            logger.info('Removing archive from host...')
+            ssh.run_command_on_host('rm {0}'.format(archive_path_on_host),
+                                    use_sudo=True,
+                                    host_string=host_string,
+                                    key_filename=ssh_key)
     else:
-        archive_path_on_manager = _archive_logs(logger)
+        host_string = env.build_manager_host_string()
+        archive_path_on_manager = _archive_logs(logger, host_string)
         if not output_path:
             output_path = os.getcwd()
         logger.info('Downloading archive to: {0}'.format(output_path))
-        ssh.get_file_from_manager(archive_path_on_manager, output_path)
+        ssh.get_file_from_host(archive_path_on_manager,
+                               output_path,
+                               host_string)
         logger.info('Removing archive from manager...')
-        ssh.run_command_on_manager(
-            'rm {0}'.format(archive_path_on_manager), use_sudo=True)
+        ssh.run_command_on_host('rm {0}'.format(archive_path_on_manager),
+                                host_string,
+                                use_sudo=True)
+
+
+def _get_cluster_nodes():
+    cluster_nodes = []
+    for nodes_list in env.profile.cluster.values():
```

Comment:
Why not just add all list and not go over one by one?

Reply:
Great point